src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -241,18 +241,27 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
             } else {
                 n_threads_per_stream = proc_type_table[0][ALL_PROC];
             }
-        } else if (hint_model_distribution_policy.size() == 0) {
-            for (auto& row : proc_socket_table) {
-                if (row[PROC_SOCKET_ID] == current_socket_id) {
-                    n_threads_per_stream = std::max(n_threads_per_stream, row[ALL_PROC]);
-                }
-            }
         } else {
-            for (size_t i = 1; i < proc_type_table.size(); i++) {
-                if (proc_type_table[i][PROC_SOCKET_ID] == current_socket_id) {
-                    n_threads_per_stream = std::max(n_threads_per_stream, proc_type_table[i][ALL_PROC]);
-                }
+            int numa_index = 1;
+            n_threads_per_stream = model_prefer_threads == 0
+                                       ? proc_type_table[numa_index][ALL_PROC]
+                                       : std::min(proc_type_table[numa_index][ALL_PROC], model_prefer_threads);
```

Comment:
Because there are 43 cores on numa node 1 and 2, 42 cores on numa node 3 on some GNR X3 machine. So infer threads may be 43 or 42, depending on which numa node  the app running on.

Reply:
Yes, it is expected behavior. OV RT uses the NUMA nodes where the application is running.